metadata/images.go

```diff
@@ -201,6 +201,22 @@ func (s *imageStore) Delete(ctx context.Context, name string) error {
 }
 
 func validateImage(image *images.Image) error {
+	if image.Name == "" {
+		return errors.Wrapf(errdefs.ErrInvalidArgument, "image name must not be empty")
+	}
+
+	if err := image.Target.Digest.Validate(); err != nil {
+		return errors.Wrapf(errdefs.ErrInvalidArgument, "Image.Target.Digest %q invalid: %v", image.Target.Digest, err)
+	}
+
+	if image.Target.Size <= 0 {
```

Comment:
Why are we validating some, not all, of the Target fields?

Reply:
These are the ones we store. I've broken these off into another function and made a note of this.